packages/expo-ui/ios/Modifiers/ViewModifierRegistry.swift

```diff
@@ -115,6 +115,92 @@ internal struct ForegroundColorModifier: ViewModifier {
   }
 }
 
+internal struct ForegroundStyleModifier: ViewModifier {
+  let styleType: String
+  let hierarchicalStyle: String?
+  let color: Color?
+  let gradientColors: [Color]?
+  let gradientStartPoint: UnitPoint?
+  let gradientEndPoint: UnitPoint?
+  let gradientCenter: UnitPoint?
+  let gradientStartRadius: CGFloat?
+  let gradientEndRadius: CGFloat?
+
+  func body(content: Content) -> some View {
+    switch styleType {
+    case "color":
+      if let color = color {
+        content.foregroundStyle(color)
+      } else {
+        content
+      }
+    case "hierarchical":
+      switch hierarchicalStyle {
+      case "primary":
+        content.foregroundStyle(.primary)
+      case "secondary":
+        content.foregroundStyle(.secondary)
+      case "tertiary":
+        content.foregroundStyle(.tertiary)
+      case "quaternary":
+        content.foregroundStyle(.quaternary)
+      case "quinary":
+        if #available(iOS 16.0, *) {
+          content.foregroundStyle(.quinary)
+        } else {
+          content.foregroundStyle(.quaternary)
+        }
+      default:
+        content.foregroundStyle(.primary)
+      }
+    case "linearGradient":
+      if let colors = gradientColors,
+        let startPoint = gradientStartPoint,
+        let endPoint = gradientEndPoint {
```

Comment:
```suggestion
      if let colors, let startPoint, let endPoint {
```
This is why I mentioned the naming